### PR TITLE
feature:add team blazor page and service (undo teamrole and teampermission).

### PR DESCRIPTION
**Pull request by MayueCif**
note  of:
1.TeamBaseInfoDto. dto whether implement interface
2.add team event handler

### CODE REVIEW

File: src/ApiGateways/Masa.Auth.ApiGateways.Caller/Services/Subjects/TeamService.cs [hunk 1]
```diff
@@ -1,44 +1,52 @@
-﻿namespace Masa.Auth.ApiGateways.Caller.Services.Subjects;
+﻿using Microsoft.AspNetCore.Mvc;
```
Comment (doddgu) [inline src/ApiGateways/Masa.Auth.ApiGateways.Caller/Services/Subjects/TeamService.cs:1]: Why using mvc?

File: src/Contracts/Masa.Auth.Contracts.Admin/Organizations/UpsertDepartmentDto.cs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 public class UpsertDepartmentDto : BaseUpsertDto<Guid>
 {
+
```
Comment (doddgu) [inline src/Contracts/Masa.Auth.Contracts.Admin/Organizations/UpsertDepartmentDto.cs:5]: Why add a blank line.

File: src/Contracts/Masa.Auth.Contracts.Admin/Subjects/TeamDetailDto.cs [hunk 3]
```diff
@@ -1,56 +1,66 @@
-﻿namespace Masa.Auth.Contracts.Admin.Subjects;
+﻿using System.ComponentModel;
```
Comment (doddgu) [inline src/Contracts/Masa.Auth.Contracts.Admin/Subjects/TeamDetailDto.cs:1]: global using

File: src/Contracts/Masa.Auth.Contracts.Admin/Subjects/UpdateTeamBaseInfoDto.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿namespace Masa.Auth.Contracts.Admin.Subjects;
+
+public class UpdateTeamBaseInfoDto
```
Comment (doddgu) [inline src/Contracts/Masa.Auth.Contracts.Admin/Subjects/UpdateTeamBaseInfoDto.cs:3]: BasicInfo

File: src/Contracts/Masa.Auth.Contracts.Admin/Subjects/UpdateTeamPersonnelDto.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+﻿namespace Masa.Auth.Contracts.Admin.Subjects;
+
+public class UpdateTeamPersonnelDto
```
Comment (doddgu) [inline src/Contracts/Masa.Auth.Contracts.Admin/Subjects/UpdateTeamPersonnelDto.cs:3]: Personnel -> Member?

File: src/Services/Masa.Auth.Service.Admin/Application/Subjects/CommandHandler.cs [hunk 6]
```diff
@@ -80,4 +83,56 @@ public async Task DeleteStaffAsync(RemoveStaffCommand deleteStaffCommand)
...
+        Team team = new Team(dto.Name, dto.Description, dto.Type, new AvatarValue(dto.Avatar.Name, dto.Avatar.Color));
+        await _teamRepository.AddAsync(team);
+        await _teamRepository.UnitOfWork.SaveChangesAsync();
+
+        await _teamDomainService.SetTeamAdminAsync(team, dto.AdminStaffs, dto.AdminRoles, dto.AdminPermissions);
+        await _teamDomainService.SetTeamMemberAsync(team, dto.MemberStaffs, dto.MemberRoles, dto.MemberPermissions);
+    }
+
+    [EventHandler]
+    public async Task UpdateTeamBaseInfoAsync(UpdateTeamBaseInfoCommand updateTeamBaseInfoCommand)
```
Comment (doddgu) [inline src/Services/Masa.Auth.Service.Admin/Application/Subjects/CommandHandler.cs:102]: BasicInfo

File: src/Services/Masa.Auth.Service.Admin/Application/Subjects/CommandHandler.cs [hunk 7]
```diff
@@ -80,4 +83,56 @@ public async Task DeleteStaffAsync(RemoveStaffCommand deleteStaffCommand)
...
+    }
+
+
+    [EventHandler]
+    public async Task RemoveTeamAsync(RemoveTeamCommand removeTeamCommand)
+    {
+        var team = await _teamRepository.GetByIdAsync(removeTeamCommand.TeamId);
+        if (team.TeamStaffs.Any())
+        {
+            throw new UserFriendlyException("the team has staffs can`t delete");
```
Comment (doddgu) [inline src/Services/Masa.Auth.Service.Admin/Application/Subjects/CommandHandler.cs:132]: Capitalize the first letter of a sentence

File: src/Services/Masa.Auth.Service.Admin/Application/Subjects/Commands/UpdateTeamBaseInfoCommand.cs [hunk 8]
```diff
@@ -0,0 +1,5 @@
+﻿namespace Masa.Auth.Service.Admin.Application.Subjects.Commands;
+
+public record UpdateTeamBaseInfoCommand(UpdateTeamBaseInfoDto UpdateTeamBaseInfoDto) : Command
```
Comment (doddgu) [inline src/Services/Masa.Auth.Service.Admin/Application/Subjects/Commands/UpdateTeamBaseInfoCommand.cs:3]: BasicInfo

File: src/Services/Masa.Auth.Service.Admin/Application/Subjects/Commands/UpdateTeamPersonnelCommand.cs [hunk 9]
```diff
@@ -0,0 +1,5 @@
+﻿namespace Masa.Auth.Service.Admin.Application.Subjects.Commands;
+
+public record UpdateTeamPersonnelCommand(UpdateTeamPersonnelDto UpdateTeamPersonnelDto, TeamMemberTypes MemberType) : Command
```
Comment (doddgu) [inline src/Services/Masa.Auth.Service.Admin/Application/Subjects/Commands/UpdateTeamPersonnelCommand.cs:3]: Member

File: src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Organization/CopyOrgSheet.razor [hunk 10]
```diff
@@ -79,7 +79,7 @@
                     <MSwitch @bind-Value="@Dto.Enabled" Inset Label="@T("Enable")" class="mt-0"></MSwitch>
                     <MSpacer />
                     <MButton class="rounded-pill" Outlined MinWidth="100" OnClick="()=> _step = 1 ">@T("Previous step")</MButton>
-                    <MButton class="rounded-pill primary" MinWidth="100" OnClick="OnSubmitHandler">@T("Submit")</MButton>
+                    <MButton class="rounded-pill primary" MinWidth="100" Disabled="@string.IsNullOrEmpty(Dto.Name.Trim())" OnClick="OnSubmitHandler">@T("Submit")</MButton>
```
Comment (doddgu) [inline src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Organization/CopyOrgSheet.razor:82]: IsNullOrWhiteSpace

**Comment by doddgu** [comment]
@capdiem

File: src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Subjects/Teams/TeamMember.razor [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+                @data.Item.Name
+            </MChip>
+        </SelectionContent>
+        <ItemContent Context="data">
+            <MListItemAvatar>
+                <img src="@data.Item.Avatar">
+            </MListItemAvatar>
+            <MListItemContent>
+                <MListItemTitle>
+                    @((MarkupString)data.Item.Name)
```
Comment (capdiem) [inline src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Subjects/Teams/TeamMember.razor:28]: MarkupString again?

File: src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Subjects/Teams/UpdateSheet.razor.cs [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+    public EventCallback<UpdateTeamPersonnelDto> OnUpdateMember { get; set; }
+
+    [Parameter]
+    public EventCallback<Guid> OnDelete { get; set; }
+
+    StringNumber _tab = 1;
+
+    protected override void OnInitialized()
+    {
+        Dto.TeamBaseInfo.PropertyChanged += (sender, args) => StateHasChanged();
```
Comment (capdiem) [inline src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Subjects/Teams/UpdateSheet.razor.cs:32]: 不建议使用PropertyChanged

**Review by capdiem — CHANGES_REQUESTED** [review]

File: src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Subjects/Teams/AddSheet.razor.cs [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+        if (OnSubmit.HasDelegate)
+        {
+            await OnSubmit.InvokeAsync(_teamDetailDto);
+        }
+        await ShowChanged.InvokeAsync(false);
+    }
+
+    protected override void OnInitialized()
+    {
+        _teamDetailDto.TeamBaseInfo.PropertyChanged += (sender, args) => StateHasChanged();
```
[the remark this answers is not in the document]
Comment (MayueCif) [reply src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Subjects/Teams/AddSheet.razor.cs]: 意见不统一 待定修改方案

File: src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Subjects/Teams/UpdateSheet.razor.cs (hunk 12 above)
Comment (MayueCif) [reply src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Subjects/Teams/UpdateSheet.razor.cs]: 意见不统一 待定修改方案

**Review by doddgu — COMMENTED** [review]
LGTM

**Review by capdiem — APPROVED** [review]
LGTM. @doddgu

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**